src/main.rs

```diff
@@ -672,7 +672,26 @@ fn exec_cargo_inner(
     if progress.count != 0 && !cx.print_command_list && cx.log_group == LogGroup::None {
         eprintln!();
     }
-    progress.count += 1;
+
+    let new_count = progress.count + 1;
+    let mut skip = false;
+    if let Some(partition) = &cx.partition {
+        if progress.count % partition.count != partition.index - 1 {
+            let mut msg = String::new();
+            if term::verbose() {
+                write!(msg, "skipping {line}").unwrap();
+            } else {
+                write!(msg, "skipping {line} on {}", cx.packages(id).name).unwrap();
+            }
+            write!(msg, " ({}/{})", new_count, progress.total).unwrap();
```

Comment:
this may be subjective, but i liked explicitly printing messages about being skipped.